metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -347,18 +347,16 @@ def gen_dataset_key(self, db_name: str, schema: str) -> PlatformKey:
             project_id=db_name,
             dataset_id=schema,
             platform=self.platform,
-            instance=self.config.platform_instance
-            if self.config.platform_instance is not None
-            else self.config.env,
+            instance=self.config.platform_instance,
+            backcompat_instance_for_guid=self.config.env,
         )
 
     def gen_project_id_key(self, database: str) -> PlatformKey:
         return ProjectIdKey(
             project_id=database,
             platform=self.platform,
-            instance=self.config.platform_instance
-            if self.config.platform_instance is not None
-            else self.config.env,
+            instance=self.config.platform_instance,
```

Comment:
As far as I know, bigquery connector does not support platform instance, because the project id is already unique.
So does platform instance make sense for big query ProjectIdKey?

Reply:
I think there's still use cases for having a platform instance with BigQuery. For example, if a company has two entirely separate GCP accounts (e.g. through an acquisition), each with their own sets of projects.

Reply:
While that may be true, it would apply to bigquery datasets too, no? So why not adding platform instance to datasets too?

As a user I would expect that when I do a search and apply filters, datasets and their containers are fully aligned.